src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLast.cs

```diff
@@ -120,5 +134,24 @@ public override void Initialize(AnalysisContext context)
                 SymbolKind.Method);
             });
         }
+
+        private static bool HasCallerInformationAttribute(IParameterSymbol parameter, ImmutableArray<INamedTypeSymbol> callerAttributes)
+        {
+            if (!parameter.IsOptional)
+            {
+                // The compiler doesn't allow caller info attributes on non-optional parameters.
+                return false;
+            }
+
+            var attributes = parameter.GetAttributes();
+            foreach (var attribute in attributes)
+            {
+                if (callerAttributes.Any(callerAttribute => SymbolEqualityComparer.Default.Equals(callerAttribute, attribute.AttributeClass)))
+                {
+                    return true;
+                }
+            }
+            return false;
```

Comment:
You could rewrite as a linq expression.

Reply:
@Evangelink that will need to use `Intersect` right? or I'm missing another better way?

Reply:
I haven't tested locally but I think the following should work `parameter.GetAttributes().Any(attribute => callerAttributes.Any(callerAttribute => SymbolEqualityComparer.Default.Equals(callerAttribute, attribute.AttributeClass)));`